Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2005-10-07
Initial Package Version: 4.0.2
Origin: Jim Gifford
Upstream Status: Not Sent
Description: Make GCC only build the 64 bit ABI's for MIPS
 
diff -Naur gcc-4.0.2.orig/gcc/config/mips/linux64.h gcc-4.0.2/gcc/config/mips/linux64.h
--- gcc-4.0.2.orig/gcc/config/mips/linux64.h	2005-01-26 02:04:46.000000000 +0000
+++ gcc-4.0.2/gcc/config/mips/linux64.h	2005-10-07 15:04:39.000000000 +0000
@@ -47,13 +47,11 @@
       %{!static: \
         %{rdynamic:-export-dynamic} \
         %{!dynamic-linker: \
-	  %{mabi=n32: -dynamic-linker /lib32/ld.so.1} \
-	  %{mabi=64: -dynamic-linker /lib64/ld.so.1} \
-	  %{mabi=32: -dynamic-linker /lib/ld.so.1}}} \
+	  %{mabi=n32: -dynamic-linker /lib/ld.so.1} \
+	  %{mabi=64: -dynamic-linker /lib64/ld.so.1}}} \
       %{static:-static}}} \
 %{mabi=n32:-melf32%{EB:b}%{EL:l}tsmipn32} \
-%{mabi=64:-melf64%{EB:b}%{EL:l}tsmip} \
-%{mabi=32:-melf32%{EB:b}%{EL:l}tsmip}"
+%{mabi=64:-melf64%{EB:b}%{EL:l}tsmip}"
 
 #undef LOCAL_LABEL_PREFIX
 #define LOCAL_LABEL_PREFIX (TARGET_OLDABI ? "$" : ".")
diff -Naur gcc-4.0.2.orig/gcc/config/mips/t-linux64 gcc-4.0.2/gcc/config/mips/t-linux64
--- gcc-4.0.2.orig/gcc/config/mips/t-linux64	2005-01-17 21:09:14.000000000 +0000
+++ gcc-4.0.2/gcc/config/mips/t-linux64	2005-10-07 06:31:48.000000000 +0000
@@ -1,6 +1,6 @@
-MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi=64
-MULTILIB_DIRNAMES = n32 32 64
-MULTILIB_OSDIRNAMES = ../lib32 ../lib ../lib64
+MULTILIB_OPTIONS = mabi=n32/mabi=64
+MULTILIB_DIRNAMES = n32 64
+MULTILIB_OSDIRNAMES = ../lib ../lib64
 
 EXTRA_MULTILIB_PARTS=crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o
 
